src/modules/sdlog2/sdlog2.c

```diff
@@ -584,6 +596,12 @@ static void *logwriter_thread(void *arg)
 		if (++poll_count == 10) {
 			fsync(log_fd);
 			poll_count = 0;
+
+			/* check if space is available, if not stop everything */
+			if (check_free_space() != OK) {
```

Comment:
Throttled to about every 10s: 04c273b.
Changed it to 10MB in abc8bf2 because you still don't want to lose logging data if you could actually coninue logging.